lib/index.coffee

```diff
@@ -0,0 +1,58 @@
+_ = require "underscore"
+_.mixin require 'underscore.deep'
+async = require "async"
+genericPool = require "generic-pool"
+
+create_pool = (thrift, options) ->
+  genericPool.Pool
+    name: "thrift"
+    create: (cb) ->
+      connection = thrift.createConnection options.host, options.port, {timeout: options.timeout}
+      connection.__ended = false
+      connection.on "connect", ->
+        connection.connection.setKeepAlive(true)
+        cb null, connection
+      connection.on "close", ->
+        connection.__ended = true
+      connection.on "error", (err) ->
+        connection.__ended = true
+        cb err
```

Comment:
is it possible that we get this callback after calling back from the "connect" event? I ask because if i remember correctly we did see this from mongo